### ISSUE TITLE
Build openhabian image in docker

### ISSUE THREAD

**Issue by EliasGabrielsson** [openhab/openhabian#697]
It would be nice to be able to build the image in docker easing development on windows or mac platforms as a complement to Vagrant. (Will conserve diskspace on host.)

Currently trying to build the image spit out this error when running: `./build.sh rpi` from the docker container.
```
supermin: failed to find a suitable kernel (host_cpu=x86_64).

I looked for kernels in /boot and modules in /lib/modules.

If this is a Xen guest, and you only have Xen domU kernels
installed, try installing a fullvirt kernel (only for
supermin use, you shouldn't boot the Xen guest with it).
libguestfs: error: /usr/bin/supermin exited with error status 1, see debug messages above
libguestfs: trace: launch = -1 (error)
libguestfs: trace: close
```

Best would be to have an `docker` option for the buildscript which invoke the build in the docker container.
When the image is ready it's copied back to host, eg. `docker cp install-test:/opt/openhabian/openhabian-pi-raspbian-201908301225-git5bf4020-crc6dcec52a.img.xz .`

**Comment by EliasGabrielsson** [comment]
@holgerfriedrich do you have any wise words about this?

**Comment by EliasGabrielsson** [comment]
This will also enable us to deprecate the Vagrantfile

**Comment by holgerfriedrich** [comment]
@EliasGabrielsson not yet, had to setup docker first, now able to reproduce. libguest seems not working in our Docker image.
Actually using the implementation with kpartx instead of guestmount (before I submitted #614) works inside my docker setup for root. This could be a used as a workaround. But I don't like implementing it in two different ways. Lets see if I can find a better way for mounting the image file....

**Comment by EliasGabrielsson** [comment]
Don't waste to much time on in was just curios if you noticed something right away. I totally agree with you that maintaining two ways of doing this would be a mess though. On the other hand is this code very stable and rarely changes so maybe we can have to implementations.

**Comment by mstormi** [comment]
Now with #682 merged and #683, #684 soon to be plus a number of improvements such as #826,
will we still want this?
If no please close.

**Comment by holgerfriedrich** [comment]
@EliasGabrielsson I do not see a possibility to do the userspace mounts in userspace in docker, so build.bash needs to be run as root. Current implementation of build.bash gives you an error message with recommendation to run the script as root in this environments. So I will close this, please feel free to reopen.
@ecdye using github actions does not solve this, it shows exactly the same problem and also needed special handling, see #852
